src/main/java/com/graphhopper/converter/core/Converter.java

```diff
@@ -1,34 +1,62 @@
 package com.graphhopper.converter.core;
 
 import com.graphhopper.converter.api.GHResponse;
-import com.graphhopper.converter.api.NominatimResponse;
+import com.graphhopper.converter.api.NominatimEntry;
+import com.graphhopper.converter.api.OpenCageDataEntry;
+import com.graphhopper.converter.api.OpenCageDataResponse;
+import com.graphhopper.converter.api.Status;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import javax.ws.rs.core.Response;
 
 /**
  * @author Robin Boldt
  */
-public class Converter
-{
+public class Converter {
 
-    public static GHResponse convertFromNominatim( NominatimResponse response )
-    {
+    public static GHResponse convertFromNominatim(NominatimEntry response) {
         return new GHResponse(response.getOsmId(), response.getLat(), response.getLon(), response.getDisplayName(), response.getAddress().getCountry(), response.getAddress().getGHCity());
     }
 
-    public static List<GHResponse> convertFromNominatimList( List<NominatimResponse> nominatimResponses )
-    {
-
-        List<GHResponse> ghResponses = new ArrayList<GHResponse>(nominatimResponses.size());
-
-        for (NominatimResponse nominatimResponse : nominatimResponses)
-        {
+    public static Response convertFromNominatimList(List<NominatimEntry> nominatimResponses, Status status) {
+        List<GHResponse> ghResponses = new ArrayList<>(nominatimResponses.size());
+        for (NominatimEntry nominatimResponse : nominatimResponses) {
             ghResponses.add(convertFromNominatim(nominatimResponse));
         }
 
-        return ghResponses;
+        return createResponse(ghResponses, status);
+    }
 
+    public static GHResponse convertFromOpenCageData(OpenCageDataEntry response) {
+        return new GHResponse(1 /*TODO*/, response.getGeometry().lat, response.getGeometry().lng,
```

Comment:
They include results from many geocoders and not all have osm id. But if just e.g. nominatim is requested they do provide this and we need to pass this.

But for the rest we need to somehow suppress printing null values...